projects/aca-content/folder-rules/src/manage-rules/manage-rules.smart-component.spec.ts

```diff
@@ -224,88 +246,205 @@ describe('ManageRulesSmartComponent', () => {
     folderRulesService.deletedRuleId$ = of(null);
     actionsService.loading$ = of(false);
 
-    const onRuleDeleteButtonClickedSpy = spyOn(component, 'onRuleDeleteButtonClicked').and.callThrough();
+    fixture.detectChanges();
 
-    const dialogResult: any = {
+    spyOn(dialog, 'open').and.returnValue({
       afterClosed: () => of(true)
-    };
-    const dialogOpenSpy = spyOn(dialog, 'open').and.returnValue(dialogResult);
+    } as MatDialogRef<boolean>);
     const deleteRuleSpy = spyOn(folderRulesService, 'deleteRule');
-    const onRuleDeleteSpy = spyOn(component, 'onRuleDelete').and.callThrough();
 
     fixture.detectChanges();
     expect(component).toBeTruthy('expected component');
 
-    const rules = debugElement.queryAll(By.css('.aca-rule-list-item'));
-    const ruleDetails = debugElement.query(By.css('aca-rule-details'));
-    const deleteRuleBtn = fixture.debugElement.nativeElement.querySelector('#delete-rule-btn');
+    const rules = unitTestingUtils.getAllByCSS('.aca-rule-list-item');
+    const ruleDetails = unitTestingUtils.getByCSS('aca-rule-details');
```

Comment:
aca-rule-details is selector of component so please use getByDirective function instead to get that element